### PR TITLE
example-backend: add apiservicebindingrequest controller

### PR DESCRIPTION
**Pull request by sttts**
Creating APIResourceExports, updating phase and conditions, and eventually (after 10min) deleting the request again.

### CODE REVIEW

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+			kubebindv1alpha1.APIServiceBindingRequestConditionExportsReady,
+			"ServiceExportNotFound",
+			conditionsapi.ConditionSeverityError,
+			"APIServiceExport %s in the service provider cluster not found",
+			req.Status.Export,
+		)
+		return nil
+	}
+
+	if c := conditions.Get(export, conditionsapi.ReadyCondition); c == nil {
```
Comment (s-urbaniak) [inline contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go:109]: where is "Ready" actually set?

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	}
+	conditions.MarkTrue(export, kubebindv1alpha1.APIServiceBindingRequestConditionExportsReady)
+
+	if req.Status.Phase == kubebindv1alpha1.APIServiceBindingRequestPhasePending {
+		if time.Since(req.CreationTimestamp.Time) > time.Minute {
+			req.Status.Phase = kubebindv1alpha1.APIServiceBindingRequestPhaseFailed
+		}
+		req.Status.Phase = kubebindv1alpha1.APIServiceBindingRequestPhaseSucceeded
+	} else if time.Since(req.CreationTimestamp.Time) > 10*time.Minute {
+		return r.deleteServiceBindingRequest(ctx, req.Namespace, req.Status.Export)
```
Comment (s-urbaniak) [inline contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go:138]: worth an event or log?

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+			c.Message,
+			req.Status.Export,
+		)
+		return nil
+	}
+	conditions.MarkTrue(export, kubebindv1alpha1.APIServiceBindingRequestConditionExportsReady)
+
+	if req.Status.Phase == kubebindv1alpha1.APIServiceBindingRequestPhasePending {
+		if time.Since(req.CreationTimestamp.Time) > time.Minute {
+			req.Status.Phase = kubebindv1alpha1.APIServiceBindingRequestPhaseFailed
```
Comment (s-urbaniak) [inline contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go:134]: we don't have a condition that reflects this timeout.

**Comment by s-urbaniak** [comment]
looks like this needs a rebase and e2e fails

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go (hunk 1 above)
Comment (sttts) [reply contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go:109]: added. Good catch.

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go (hunk 2 above)
Comment (sttts) [reply contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go:138]: yes, all actions should be logged.

We don't have events yet. Worth an issue.

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go (hunk 3 above)
Comment (sttts) [reply contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go:134]: it is deleted. No use for a condition :)

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go (hunk 2 above)
Comment (s-urbaniak) [reply contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go]: will add 👍

File: contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go (hunk 3 above)
Comment (s-urbaniak) [reply contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go]: lol, indeed 🙈

**Comment by sttts** [comment]
Subsumed by https://github.com/kube-bind/kube-bind/pull/87.

**Pull request closed without merging**